src/pms/core/reader.py

```diff
@@ -71,6 +96,24 @@ class SensorReader(Reader):
     Valid messages are extracted from the serial buffer.
     """
 
+    @dataclass
+    class _Reading:
+        """Represents a single sensor reading
+
+        This class is only used by internal functions as a way of passing reading data between them.
+        """
+
+        buffer: bytes
+        obs_data: ObsData
+
+        @property
+        def raw_data(self) -> RawData:
+            return RawData(self.time, self.buffer)
+
+        @property
+        def time(self) -> int:
+            return self.obs_data.time
```

Comment:
Is this internal class really needed?

Reply:
I use classes because they give meaning to the data being passed around. For example, what is a `(buffer, obs_data)`? It's not obvious without a class to clarify that it represents the data from a `Reading`.

That said, I don't feel strongly about it as this is an internal method. We could have this instead:

```python
def _read_one(...):
    ...
    return (buffer, obs_data)

...

def __call__(...):
    ...
    (buffer, obs_data) = self._read_one(...)
    ...
    yield RawData(obs_data.time, buffer) if raw else obs_data
```